### DISCUSSION TITLE
Google Authentication integration

### DISCUSSION THREAD

**Discussion by karanmaster25** [directus/directus#3799] [Archive]
I want to integrate Google authentication in my project. I have added following keys in env file

OAUTH_PROVIDERS
OAUTH_GOOGLE_KEY
OAUTH_GOOGLE_SECRET
OAUTH_GOOGLE_SCOPE

Now how to enable google auth support in directus and integrate in my project because couldn't find proper documentation for google authentication.

**Comment by dboyadzhiev** [comment]
First, you have to create an OAuth 2.0 Client ID: https://console.cloud.google.com/apis/credentials with "Authorized redirect URIs": `https://<your_domain>/auth/oauth/google/callback`
From that, you will get `OAUTH_GOOGLE_KEY` (Client ID) and `OAUTH_GOOGLE_SECRET` (Client Secret)

And 
```
OAUTH_PROVIDERS = "google"
OAUTH_GOOGLE_SCOPE = "openid https://www.googleapis.com/auth/userinfo.email"
```

NB: first you have to create a user in Directus before it can log in via google
